### PR TITLE
ADD extract-storybook bin

### PR DESCRIPTION
**Pull request by ndelangen**
Storybook Composition needs a static extracted data object of the storybook stories for best performance.

I added a command the the CLI to generate such file.

`stories.json` will contain something like:

```json
{
  "stories": {
    "story-id": {
      "name": "story name",
      "kind": "story kind"
    }
  }
}
```

We might add more data in the future, so this is why the data is in a stories property.

Potential future data:

- global args
- global parameters
- config
- addons

### CODE REVIEW

**Comment by tmeasday** [comment]
Let's discuss this -- I don't think we can add `puppeteer` as a dependency of the CLI. We should probably make this a separate CLI package that folks can optionally install if they want to publish their SB for composition.

**Review by shilman — CHANGES_REQUESTED** [review]
I agree with @tmeasday that the `puppeteer` dependency makes this more suitable to a separate package. It's an awesome feature tho!!

**Comment by ndelangen** [comment]
OK, I'll extract this into it's own package

File: lib/cli/src/extract.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    await page.evaluate(async () => {
+      // eslint-disable-next-line no-undef
+      const d = window.__STORYBOOK_STORY_STORE__.extract();
+
+      const result = Object.entries(d).reduce(
+        (acc, [k, v]) => ({
+          ...acc,
+          [k]: {
+            ...v,
+            parameters: {
```
Comment (shilman) [inline lib/cli/src/extract.js]: we might need to revisit this filtering of parameters in the future, but ok for now

**Review by shilman — APPROVED** [review]
AWESOME! The lazy loading is a sweet touch 😘

**Pull request merged**